### DISCUSSION TITLE
How can I run a CodeQL query on wider github database of Public Repos?

### DISCUSSION THREAD

**Discussion by abdulhannanali** [github/securitylab#341] [Q&A] [answered]
Hi there Github Security Lab,

I am a dev who mostly sticks to development and research related to it, but have been dipping my toes a bit into security for a few days now. I encountered CodeQL and it is very cool. 

I have a CodeQL query in mind for some security research and in order to extract meaningful results I would like to run it on a large database of open source repos. I see that lgtm.com allows it to run on a few selected repos, but how can I run it on a larger number of projects? 

One of the way I envision it is to run the query sequentially for a large number of repos by adding them alternatively, would that be abuse of lgtm.com free services?

**Comment by criemen** [comment, marked as answer]
Hi @abdulhannanali ,

LGTM has project lists to which you can add a couple of thousands of projects, and then you can run queries across them.
Does this satisfy your need?
If you have more questions about LGTM, please join the github security lab slack instance, see https://securitylab.github.com/get-involved/.
There's a bunch of active security researchers who use LGTM.com for their research, so you might benefit from talking to them.

**Reply by abdulhannanali** [reply]
@criemen Adding projects to the list sounds good. I will be joining the Github Security Lab's slack as well, it sounds like a great idea. Thank you :)

**Reply by JLLeitschuh** [reply]
A few of us have written scripts to let you bulk import hundreds or thousands of repositories into your account so you can query them.

https://github.com/JLLeitschuh/lgtm_hack_scripts

**Reply by abdulhannanali** [reply]
Thank you @JLLeitschuh That seems like plenty good :heart:
